src/content/1.7/modules/concepts/controllers/admin-controllers/override-decorate-controller.md

```diff
@@ -0,0 +1,171 @@
+---
+title: Override or decorate a Core controller
+---
+
+# Override or decorate a Core controller
+
+{{< minver v="1.7.7" title="true" >}}
+
+If you want to extend the behavior of a Back Office page in PrestaShop, you have multiple options.
+
+Most of standard extension needs can be fulfilled using one [hook][hooks].
+
+If there is no hook available for your need, and you only need to modify the visual appearance of the page, you might want to [override a template][template-override].
+
+But sometimes you want to modify the page in a deeper way. In this case, you need to modify the controller's behavior.
+
+If the Back Office page you want to modify is powered by Symfony, you have 3 options:
+
+- Remap the route to target a new Controller you create
+- Override the existing Core controller
+- Decorate the existing Core controller
+
+## Remap the route
+
+In Symfony, [routes](https://symfony.com/doc/3.4/routing.html) such as `/sell/orders/orders` are mapped to controllers by YAML configuration files such as this one:
+```yaml
+# src/PrestaShopBundle/Resources/config/routing/admin/sell/orders/orders.yml
+admin_orders_index:
+  path: /sell/orders/orders/
+  methods: [GET]
+  defaults:
+    _controller: PrestaShopBundle:Admin/Sell/Order/Order:index
+    _legacy_controller: AdminOrders
+    _legacy_link: AdminOrders
+```
+
+In your module, you have the possibility to replace this configuration item by your own.
+
+So for example, if you want that people browsing the Back Office on URL `/sell/orders/orders/` to hit your own controller `MyModule\Controller\DemoController` instead of the Core's, you can do this:
+```yaml
+# modules/your-module/config/routes.yml
+admin_orders_index:
+    path: /sell/orders/orders/
+    methods: [GET]
+    defaults:
+      _controller: 'MyModule\Controller\DemoController::demoAction'
+```
+
+{{% notice tip %}}
+In the above example, the `path` has not been changed however you can change it to whatever you want like `/demo/orders` ! This however will re-route internal links but external or hardcoded links will keep targeting the old path.
```

Comment:
```suggestion
In the above example, the `path` has not been changed, but you can change it to whatever you want (eg. `/demo/orders`). However, keep in mind that while this will re-route internal links, external or hardcoded links will keep targeting the old path.
```